src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 8;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private Object[] array = new Object[ARRAY_SIZE];
+    private int counter = 0;
+
+    private void resize(int minSize) {
+        if (minSize < 0) {
+            throw new OutOfMemoryError();
+        }
+        int newSize = minSize + (minSize >> 2);
+        if (newSize - MAX_ARRAY_SIZE >= 0) {
+            throw new RuntimeException("Array is out of integer bounds and "
```

Comment:
In the rest of your implementation, you are using `ArrayIndexOutOfBoundsException`. Why here you are using `RuntimeException`?

Reply:
Since it is impossible to have < 0 array size I thought it would be suitable exception, since it's not user's fault but programmer (me) that didn't created implementation to go beyond integer size.